packages/block-editor/src/components/rich-text/event-listeners/paste-handler.js

```diff
@@ -37,10 +38,15 @@ export default ( props ) => ( element ) => {
 			return;
 		}
 
-		const { plainText, html } = getPasteEventData( event );
+		let { plainText, html } = getPasteEventData( event );
 
 		event.preventDefault();
 
+		// When line breaks are disabled, convert all line breaks to a single space.
+		plainText = disableLineBreaks
+			? plainText.replace( /\r?\n/g, ' ' ).trim()
+			: plainText;
```

Comment:
What if you paste HTML? This doesn't look like a complete solution. Could you modify `valueToInsert` within `pasteInline` instead? We should strip out all the `\n` there. There's a rich text replace function I believe.

Reply:
Hi @ellatrix,
Thank you for the thoughtful review. I've implemented the solution by modifying `valueToInsert` within `pasteInline` using the rich-text `replace` function as you recommended.

Could you please review the changes